run.py

```diff
@@ -0,0 +1,76 @@
+
+NUMBER_OF_SAMPLES:int = 5
+EXPERIMENT_TIME:timedelta = timedelta(hours=5)
+STEP_THROUGH_TIME:timedelta = timedelta(seconds=5)
+CYCLE_SLEEP_TIME:float = 0
+
+def model(display:bool, folder:str, samples:List[SampleData], random_samples:bool, number_of_samples:int,
+          experiment_time:timedelta, step_through_time:timedelta,
+          cycle_sleep_time:float) -> AMI:
+    """Run CXI Simulation"""
+    filename = str(time())
+    os.makedirs(os.path.dirname(f"results{folder}/model/r{filename}.tsv"), exist_ok=True)
+    os.makedirs(os.path.dirname(f"results{folder}/data/r{filename}.tsv"), exist_ok=True)
+    with open(f"results{folder}/model/r{filename}.tsv", "w", encoding="utf-8") as file:
+        with open(f"results{folder}/data/r{filename}.csv", "w", encoding="utf-8") as data_file:
+            context:Context = Context(
+                AMI(samples, random_samples), Agenda(experiment_time, number_of_samples),
+                DataAnalyst(), ExperimentManager(), Operator(),
+                CXI(), CommunicationObject(), file, data_file)
+            context.file.write(f"{goal_agenda_plan(context)}\n")
+            while experiment_is_not_over(context):
+                os.system('cls' if os.name == 'nt' else 'clear')
+                context.messages.reset()
+                print(goal_agenda_plan(context) if display else 'running')
+                context.update()
+                if not context.agenda.is_started():
+                    context.agent_em.start_experiment(context)
+                elif context.agenda.is_started():
+                    if not context.instrument.is_running():
+                        context.agent_em.start_instrument(context)
+                    elif context.instrument.is_running():
+                        if not context.instrument.is_collecting_data():
+                            if context.agent_em.check_if_next_run_can_be_started(context):
+                                context.agent_em.tell_operator_start_data_collection(context)
+                        elif context.instrument.is_collecting_data():
+                            context.agent_op.track_stream_position(context)
+                            context.agent_em.check_if_data_is_sufficient(context)
+                print(context.messages if display else 'running')
+                context.agent_da.check_if_experiment_is_compleated(context)
+                context.current_time += step_through_time
+                sleep(cycle_sleep_time)
+            os.system('cls' if os.name == 'nt' else 'clear')
+            context.file.write(f"\n{experiment_stats(context)}\nFinished")
+            print(f"{experiment_stats(context)}\n{colored('Finished','green')}" if display else '--')
+            create_experiment_figure(context, False)
+            return copy.deepcopy(context.ami)
+# model(True, '', [], True, NUMBER_OF_SAMPLES, EXPERIMENT_TIME, STEP_THROUGH_TIME, CYCLE_SLEEP_TIME)
```

Comment:
If you want to run just this file uncomment this line, otherwise if you are running through the jig leave it commented